### PR TITLE
Update namespaces page

### PR DESCRIPTION
**Pull request by YashPimple**

### CODE REVIEW

**Comment by sftim** [comment]
/retitle Update namespaces page

Please also fill in the PR description to explain why you are making this change. Then, reviewers can look at the change you're making and see how well those changes match up with the need for a fix.

File: content/en/docs/concepts/workloads/pods/user-namespaces.md [hunk 1]
```diff
@@ -89,10 +89,10 @@ isolate the users in the container from the users in the node.
...
 host. Inside the container the process will think it is running as root (and
 therefore tools like `apt`, `yum`, etc. work fine), while in reality the process
-doesn't have privileges on the host. You can verify this, for example, if you
-check the user the container process is running `ps` from the host. The user
-`ps` shows is not the same as the user you see if you execute inside the
-container the command `id`.
+doesn't have privileges on the host. You can verify it from the host by checking the user's 
+container in which the process `ps` is running. The user `ps` shows is not the same as the user 
+you see if the command `id` is execute inside the container. The command `ps` command doesn't 
+show the user ID or user name.
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/user-namespaces.md:95]: Still, this text is confusing. If the `ps` command doesn't show the user ID or the user name, what does "The user `ps` shows ..." mean?

File: content/en/docs/concepts/workloads/pods/user-namespaces.md [hunk 2]
```diff
@@ -89,10 +89,10 @@ isolate the users in the container from the users in the node.
 This means containers can run as root and be mapped to a non-root user on the
 host. Inside the container the process will think it is running as root (and
 therefore tools like `apt`, `yum`, etc. work fine), while in reality the process
-doesn't have privileges on the host. You can verify this, for example, if you
-check the user the container process is running `ps` from the host. The user
-`ps` shows is not the same as the user you see if you execute inside the
-container the command `id`.
+doesn't have privileges on the host. You can verify it from the host by checking the user's
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/workloads/pods/user-namespaces.md:92]:
```suggestion
doesn't have privileges on the host. You can verify this, for example, if you
```

File: content/en/docs/concepts/workloads/pods/user-namespaces.md [hunk 3]
```diff
@@ -89,10 +89,10 @@ isolate the users in the container from the users in the node.
 This means containers can run as root and be mapped to a non-root user on the
 host. Inside the container the process will think it is running as root (and
 therefore tools like `apt`, `yum`, etc. work fine), while in reality the process
-doesn't have privileges on the host. You can verify this, for example, if you
-check the user the container process is running `ps` from the host. The user
-`ps` shows is not the same as the user you see if you execute inside the
-container the command `id`.
+doesn't have privileges on the host. You can verify it from the host by checking the user's 
+container in which the process `ps` is running. The user `ps` shows is not the same as the user
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/workloads/pods/user-namespaces.md:93]:
```suggestion
check the user that the container process is running as by executing `ps` on the host. The user
```

File: content/en/docs/concepts/workloads/pods/user-namespaces.md [hunk 4]
```diff
@@ -89,10 +89,10 @@ isolate the users in the container from the users in the node.
 This means containers can run as root and be mapped to a non-root user on the
 host. Inside the container the process will think it is running as root (and
 therefore tools like `apt`, `yum`, etc. work fine), while in reality the process
-doesn't have privileges on the host. You can verify this, for example, if you
-check the user the container process is running `ps` from the host. The user
-`ps` shows is not the same as the user you see if you execute inside the
-container the command `id`.
+doesn't have privileges on the host. You can verify it from the host by checking the user's 
+container in which the process `ps` is running. The user `ps` shows is not the same as the user 
+you see if the command `id` is execute inside the container. The command `ps` command doesn't
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/workloads/pods/user-namespaces.md:94]:
```suggestion
in the output of the `ps` command is not the same if you execute the command `id` inside the
```

File: content/en/docs/concepts/workloads/pods/user-namespaces.md (hunk 1 above)
Comment (divya-mohan0209) [inline content/en/docs/concepts/workloads/pods/user-namespaces.md:95]:
```suggestion
container.
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
I've phrased this a bit better. WDYT?

**Comment by YashPimple** [comment]
> I've phrased this a bit better. WDYT?

Sure I would implement it.

**Comment by dilyanpalauzov** [comment]
The documentation of user-namespaces by CRI-O is described at https://github.com/cri-o/cri-o/blob/main/tutorials/userns.md . It says that the user `containers` must exist on the host, and it must be present in the `/etc/subuid` and `/etc/subgid` files.  The user namespace mode is utilized by pods with the `io.kubernetes.cri-o.userns-mode` annotation.  User-namespace pods can mount `Directory` volumes.

The description at https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/ does not say under which host user are the pods running (cf. CRI-O which uses the `containers` user) and it states that `Directory` voluimes cannot be used.

Please extend both https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/ and the upcoming blog entry 2022-10-03-add-userns-alpha with information about the host user, under which the containers will be running and state, if the user must be in the ` /etc/subuid` and `/etc/subgid` files.

Why can’t a `hostUsers: false` pod use a Directory volume, but a Pod with the annotation `io.kubernetes.cri-o.userns-mode` can?

**Comment by sftim** [comment]
@dilyanpalauzov are you asking that @YashPimple update the upcoming blog entry? That seems a big ask. Maybe you'd be willing to file an issue that records the changes you'd like to see.

**Comment by dilyanpalauzov** [comment]
I am saying what currently is unclear to me.  If my hint is not sufficient to initiate improvements on the documentation, I doubt that any more formal requests will achieve it.

The missing peaces are both relevant to https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/ and the upcoming blog entry.

**Comment by sftim** [comment]
> I am saying what currently is unclear to me. If my hint is not sufficient to initiate improvements on the documentation, I doubt that any more formal requests will achieve it.

In this instance @dilyanpalauzov I'm confident you're wrong - for open source, filing an issue is a great way to flag work up to the volunteers who might pick it up. It sounds like you wouldn't be willing to in this case, and that's OK.

**Comment by dilyanpalauzov** [comment]
> In this instance @dilyanpalauzov I'm confident you're wrong - for open source, filing an issue is a great way to flag work up to the volunteers who might pick it up. It sounds like you wouldn't be willing to in this case, and that's OK.

In case of Kubernetes, most of the issues I opened were automatically closed after three months of inactivity.

**Comment by Sea-n** [comment]
I've see you opened PR #36738 and #36867 for the same fix, close the old one as duplicated.
/close

**Pull request closed without merging**